tls/extensions/s2n_server_supported_versions.c

```diff
@@ -38,7 +38,7 @@
  * Selected Version (2 byte)
  **/
 
-int s2n_extensions_server_supported_versions_size()
+int s2n_extensions_server_supported_versions_size(struct s2n_connection *conn)
```

Comment:
Why add the connection struct here?

Reply:
i'm refactoring this to take in a conn to aline with all  `*_send_sizes(conn)` so we can have a unified extension interface.

Reply:
mark as unused so as not to generate warnings?  `-Wunused-parameter`

````c
__attribute__(unused)